src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,192 @@
+package frc.robot.leds;
+
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import java.awt.*;
+
+public class LedStrip extends SubsystemBase {
+    private final AddressableLED ledStrip;
+    private AddressableLEDBuffer ledBuffer;
+
+    private double blinkTime = 1;
+    private double fadeDuration = 1;
+    private int rainbowHue = 0;
+
+    private Color primary = new Color(0);
+    private Color secondary = new Color(0);
+    private Color currentColor = primary;
```

Comment:
I believe this is something that can be managed inside each mode handling function. Usually if you have state that is relevant to one method only, it's better to store that state in that method only.

Reply:
I agree but only about `rainbowHue`, the others are needed in the periodic method.

Reply:
In a quick glance it looks like they are used only in the periodic method. Perhaps changing the logic to use a command will be better? That way the "periodic" code will be the command's `execute()` method.

Reply:
I don't think it's necessary, it'll just be confusing.